messaging/src/main/java/org/axonframework/messaging/DelayedMessageStream.java

```diff
@@ -75,13 +75,61 @@ public CompletableFuture<Entry<M>> firstAsCompletableFuture() {
 
     @Override
     public Flux<Entry<M>> asFlux() {
-        return Mono.fromFuture(delegate)
-                   .flatMapMany(MessageStream::asFlux);
+        return Mono.fromFuture(delegate).flatMapMany(MessageStream::asFlux);
     }
 
     @Override
-    public <R> CompletableFuture<R> reduce(@Nonnull R identity,
-                                           @Nonnull BiFunction<R, Entry<M>, R> accumulator) {
+    public Optional<Entry<M>> next() {
+        if (delegate.isDone()) {
+            return delegate.getNow(null).next();
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void onAvailable(@Nonnull Runnable callback) {
+        delegate.whenComplete((r, e) -> {
+            if (r != null) {
+                r.onAvailable(callback);
+            } else {
+                callback.run();
```

Comment:
The error is captured in the delegate itself. This is just to notify the callback that a value is available for reading. In an error case, it would be an indicator that the isComplete() flag is set and the error() contains a value.

Reply:
But of course. Thanks for clarifying.